quickwit/quickwit-indexing/src/actors/doc_processor.rs

```diff
@@ -43,16 +48,46 @@ use crate::models::{
 
 const PLAIN_TEXT: &str = "plain_text";
 
+#[derive(Debug)]
+pub enum DocProcessorError {
+    MissingField,
+    ParsingError,
+    #[cfg(feature = "vrl")]
+    TransformError(VrlTerminate),
+}
+
+impl From<serde_json::Error> for DocProcessorError {
+    fn from(_error: serde_json::Error) -> Self {
+        DocProcessorError::ParsingError
+    }
+}
+
+impl From<FromUtf8Error> for DocProcessorError {
+    fn from(_error: FromUtf8Error) -> Self {
+        DocProcessorError::ParsingError
+    }
+}
+
+impl From<OtlpTraceError> for DocProcessorError {
+    fn from(_error: OtlpTraceError) -> Self {
+        DocProcessorError::ParsingError
+    }
+}
+
 enum InputDoc {
     Json(Bytes),
+    OtlpTraceJson(Bytes),
+    OtlpTraceProtobuf(Bytes),
     PlainText(Bytes),
 }
 
 impl InputDoc {
-    fn from_bytes(input_format: &SourceInputFormat, bytes: Bytes) -> Self {
+    fn from_raw_doc(input_format: SourceInputFormat, raw_doc: Bytes) -> Self {
         match input_format {
-            SourceInputFormat::Json => InputDoc::Json(bytes),
-            SourceInputFormat::PlainText => InputDoc::PlainText(bytes),
+            SourceInputFormat::Json => InputDoc::Json(raw_doc),
```

Comment:
I think the `enum` on `InputDoc` is not helpful if we end up keeping the raw payload anyway.

```rust
struct InputDoc {
    input_format: SourceInputFormat,
    raw_doc: Bytes,
}
```